relayer/src/util/retry.rs

```diff
@@ -2,9 +2,33 @@ use std::time::Duration;
 
 pub use retry::{
     delay::{Fibonacci, Fixed},
-    retry_with_index, OperationResult as RetryResult,
+    retry_with_index, Error as RetryError, OperationResult as RetryResult,
 };
 
+pub trait ToRetryResult<T, E> {
+    fn err_to_retry_result(self, cont: impl FnOnce(E) -> RetryResult<T, E>) -> RetryResult<T, E>;
+}
+
+impl<T, E> ToRetryResult<T, E> for Result<T, E> {
+    fn err_to_retry_result(self, cont: impl FnOnce(E) -> RetryResult<T, E>) -> RetryResult<T, E> {
+        match self {
+            Ok(res) => RetryResult::Ok(res),
+            Err(err) => cont(err),
+        }
+    }
+}
+
+pub fn count_retry<E>(err: RetryError<E>) -> u64 {
```

Comment:
Reads a little bit better imho

```suggestion
pub fn retry_count<E>(err: RetryError<E>) -> u64 {
```

Reply:
Done.